lib/rubocop/cop/packs/zeitwerk_friendly_constant.rb

```diff
@@ -0,0 +1,164 @@
+# typed: false
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Packs
+      # This cop checks that every constant defined in a file matches the
+      # file name such that it is independently loadable by Zeitwerk.
+      #
+      # @example
+      #
+      #   Good
+      #
+      #     # /some/directory/foo.rb
+      #     module Foo
+      #     end
+      #
+      #     # /some/directory/foo.rb
+      #     module Foo
+      #       module Bar
+      #       end
+      #     end
+      #
+      #     # /some/directory/foo/bar.rb
+      #     module Foo
+      #       module Bar
+      #       end
+      #     end
+      #
+      #   Bad
+      #
+      #     # /some/directory/foo.rb
+      #     module Bar
+      #     end
+      #
+      #     # /some/directory/foo/bar.rb
+      #     module Foo
+      #       module Bar
+      #       end
+      #
+      #       module Baz
+      #       end
+      #     end
+      #
+      class ZeitwerkFriendlyConstant < Cop
```

Comment:
Oh okay, thanks for clarifying!

Reply:
I'm not totally sure we'll be able to easily have multiple licenses present in this repo, let me know if you have any experience in this. If I can't figure it out I'll just close this PR.

Reply:
It seems this repo is also using MIT, so the licences are compatible. The way I'd do this would be just to copy the license of the gist to the top of this file.

Another way is to change your LICENSE file to add a note in the bottom saying that copyright of this file is from Shopify Inc.